### DISCUSSION TITLE
behind-the-implementation-of-dotnets-priorityqueue/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#413] [Announcements]
# behind-the-implementation-of-dotnets-priorityqueue/

In this post I show how .NET's PriorityQueue implements its methods and relate these methods to operations on a min-heap data structure

https://andrewlock.net/behind-the-implementation-of-dotnets-priorityqueue/

**Comment by NielsPilgaard** [comment]
Excellent explanations, thanks for sharing 👍

**Comment by wangengzheng** [comment]
I'm impressed by your post. It's well-written and thought-provoking

**Comment by InternalHigh** [comment]
You are obsessed by priority queues

**Comment by ihorwill** [comment]
Loving the level of detail in this series.

This comment for Dequeue seems incorrect: the test actually indicates there are no nodes left:
```cs
// If lastNodeIndex == 0, then there's only one node left, so
// there's no more work to do
```

**Reply by andrewlock** [reply]
`lastNodeIndex == 0` means there's no more nodes to swap, but there's still one node left!
